tests/view/rooms/list/app.js

```diff
@@ -0,0 +1,57 @@
+/**
+ * Copyright (c) 2018, salesforce.com, inc.
+ * All rights reserved.
+ * Licensed under the BSD 3-Clause license.
+ * For full license text, see LICENSE.txt file in the repo root or
+ * https://opensource.org/licenses/BSD-3-Clause
+ */
+
+/**
+ * tests/view/rooms/list/app.js
+ */
+
+import ReactTestUtils from 'react-dom/test-utils';
+import { jsdom } from 'jsdom'
+
+const expect = require('chai').expect;
+
+const app = require('../../../../view/rooms/list/app.js');
+
+describe('tests/view/rooms/list/app.js', () => {
+  const testArr = ['1', '2', '3'];
+
+  it('ok, select element created with 3 options', () => {
+    const selectEl = app.createSelectEl(testArr)
+    expect(selectEl.tagName).to.equal('SELECT');
+    expect(selectEl.childElementCount).to.equal(testArr.length);
+    expect(selectEl.value).to.equal('1');
+  });
+
+  it('ok, select element created and default value is overwritten', () => {
+    const selectEl = app.createSelectEl(testArr, '2')
+    expect(selectEl.value).to.equal('2');
+  });
+
+  it('ok, constructing url after page filter change', () => {
+    const url = app.constructListFilterUrl('page', '3');
+    expect(url).to.equal('/rooms?page=3');
+  });
+
+  it('ok, constructing url after type filter change (page is reset to 1)', () => {
```

Comment:
must reset page to 1 when changing the type filter?

Reply:
if skiping the previous test, would this one pass? Tests should be independent.

Reply:
@cassiodias yeah we need to reset page to 1 when the type/active filter is changed as there may be 0 results. There is no state being stored in the function so all tests are independent